rollout/trafficrouting/istio/istio_test.go

```diff
@@ -391,6 +392,52 @@ func rolloutWithDestinationRule() *v1alpha1.Rollout {
 	}
 }
 
+// TestUpdateHashWithListers verifies behavior of UpdateHash when using informers/listers
+func TestUpdateHashAdditionalFieldsWithListers(t *testing.T) {
+	ro := rolloutWithDestinationRule()
+	obj := unstructuredutil.StrToUnstructuredUnsafe(`
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: istio-destrule
+  namespace: default
+spec:
+  subsets:
+  - name: stable
+    labels:
+      version: v3
+  - name: canary
+    trafficPolicy:
+      loadBalancer:
+        simple: ROUND_ROBIN
+`)
```

Comment:
Can we improve the test to verify we also preserve fields at the `.spec.XXX` level? For example, this example from istio docs which has `spec.host` and `spec.trafficPolicy`:

```yaml
apiVersion: networking.istio.io/v1beta1
kind: DestinationRule
metadata:
  name: bookinfo-ratings
spec:
  host: ratings.prod.svc.cluster.local
  trafficPolicy:
    loadBalancer:
      simple: LEAST_CONN
  subsets:
  - name: testversion
    labels:
      version: v3
    trafficPolicy:
      loadBalancer:
        simple: ROUND_ROBIN
```

Reply:
Sure. Let me update the PR to generalize the method to the `.spec`.

Reply:
@jessesuen , I think the current implement has been able to preserve the fields of `.spec.XXX`. Following is a snippet of a destination rule after an update to a rollout.

```
spec:
  host: istio-rollout                                                <--------- Preserved
  subsets:
  - labels:
      app: istio-rollout
      rollouts-pod-template-hash: 6646844db8
    name: canary
  - labels:
      app: istio-rollout
      rollouts-pod-template-hash: cf89dfb88
    name: stable
    trafficPolicy:                                                      <--------- Preserved by the PR
      loadBalancer:
        simple: ROUND_ROBIN
  trafficPolicy:                                                         <--------- Preserved
    loadBalancer:
      simple: LEAST_CONN
```

However, I could not find the relevant part in the code. Do you have any idea?

Reply:
So to verify we preserve the fields under `.spec`, I added a string comparison between the expected result and the unstructured object.